### PR TITLE
Refactor FileField

### PR DESCRIPTION
**Pull request by thedrow**
This is a minor refactor done for performance optimization.

### CODE REVIEW

File: factory/django.py [hunk 1]
```diff
@@ -202,26 +202,29 @@ def _make_data(self, params):
...
     def _make_content(self, params):
         path = ''
+        
+        from_path = params.get('from_path')
+        from_file = params.get('from_file')
+        from_func = params.get('from_func')
 
-        _content_params = [params.get('from_path'), params.get('from_file'), params.get('from_func')]
-        if len([p for p in _content_params if p]) > 1:
+        if sum(bool(p) for p in (from_path, from_file, from_func)) > 1:
```
Comment (kingbuzzman) [inline factory/django.py:210]: For the sake of "optimization" I suggest you change it to this: (which is VERY similar to what was there before.)

```suggestion
        if len([p for p in (from_path, from_file, from_func) if p]) > 1:
```

Here are my tests, and the `timeit` results to back it up ;)
```python
import timeit



def test_old_method(from_path, from_file, from_func, expected):
    _content_params = [from_path, from_file, from_func]
    assert ((len([p for p in _content_params if p]) > 1) is expected), (from_path, from_file, from_func, expected)


def test_thedow_method(from_path, from_file, from_func, expected):
    assert (sum(bool(p) for p in (from_path, from_file, from_func)) > 1) is expected, (from_path, from_file, from_func, expected)


def test_new_method(from_path, from_file, from_func, expected):
    assert ((len([p for p in (from_path, from_file, from_func) if p]) > 1) is expected), (from_path, from_file, from_func, expected)


options = [
    # from_path, from_file, from_func, expected
    (None, None, None, False),
    (None, None, '', False),
    (None, None, '/file/path', False),
    (None, '', None, False),
    (None, '', '', False),
    (None, '', '/file/path', False),
    (None, '/file/path', None, False),
    (None, '/file/path', '', False),
    (None, '/file/path', '/file/path', True),
    ('', None, None, False),
    ('', None, '', False),
    ('', None, '/file/path', False),
    ('', '', None, False),
    ('', '', '', False),
    ('', '', '/file/path', False),
    ('', '/file/path', None, False),
    ('', '/file/path', '', False),
    ('', '/file/path', '/file/path', True),
    ('/file/path', None, None, False),
    ('/file/path', None, '', False),
    ('/file/path', None, '/file/path', True),
    ('/file/path', '', None, False),
    ('/file/path', '', '', False),
    ('/file/path', '', '/file/path', True),
    ('/file/path', '/file/path', None, True),
    ('/file/path', '/file/path', '', True),
    ('/file/path', '/file/path', '/file/path', True)
]

def full_test_old_method():
    for from_path, from_file, from_func, expected in options:
        test_old_method(from_path, from_file, from_func, expected)

def full_test_thedow_method():
    for from_path, from_file, from_func, expected in options:
        test_thedow_method(from_path, from_file, from_func, expected)

def full_test_new_method():
    for from_path, from_file, from_func, expected in options:
        test_new_method(from_path, from_file, from_func, expected)


print(timeit.timeit('full_test_old_method()', 'from __main__ import full_test_old_method', number=100000), '<-- this one is what was removed')

print(timeit.timeit('full_test_thedow_method()', 'from __main__ import full_test_thedow_method', number=100000), '<-- this is what you currently have')

print(timeit.timeit('full_test_new_method()', 'from __main__ import full_test_new_method', number=100000), '<-- this one is the suggestion')
```

```
1.0798860000213608 <-- this one is what was removed
2.2313669999712147 <-- this is what you currently have
1.03782600001432 <-- this one is the suggestion
```

**Review by kingbuzzman — APPROVED** [review]
lgtm — ps, out of curiosity, what prompted this?

**Comment by thedrow** [comment]
I just tested perflint and it complained about it so I decided to fix it real quick.

**Comment by francoisfreitag** [comment]
Docs issue should be fixed with #937.

**Pull request merged**